api/registrations/serializers.py

```diff
@@ -350,16 +353,22 @@ def update(self, registration, validated_data):
             except NodeStateError as err:
                 raise Conflict(err.message)
         if 'custom_citation' in validated_data:
-            registration.update_custom_citation(validated_data.pop('custom_citation'), auth)
+            if registration.has_permission(user, permissions.ADMIN):
+                registration.update_custom_citation(validated_data.pop('custom_citation'), auth)
+            else:
+                raise exceptions.PermissionDenied()
         is_public = validated_data.get('is_public', None)
         if is_public is not None:
             if is_public:
-                try:
-                    registration.update(validated_data, auth=auth)
-                except NodeUpdateError as err:
-                    raise exceptions.ValidationError(err.reason)
-                except NodeStateError as err:
-                    raise exceptions.ValidationError(err.message)
+                if registration.has_permission(user, permissions.ADMIN):
```

Comment:
Minor: could this be assigned to an `is_admin` variable at the beginning of this function? It would prevent us from checking permissions twice if someone updated the registration's custom citation and privacy in the same request.